v2/managedsettling.go

```diff
@@ -122,15 +133,18 @@ func NewManagedSettlingHandler(opts *ManagedSettlingOptions, handler ManagedSett
 }
 
 func (m *ManagedSettler) handleError(ctx context.Context, settler MessageSettler, message *azservicebus.ReceivedMessage, handleErr error) {
+	if handleErr == nil {
+		handleErr = fmt.Errorf("nil error: %w", handleErr)
```

Comment:
`handleErr = fmt.Errorf("nil error")`

(the %w is unnecessary, you know it's nil)